library/src/main/scala/g8.scala

```diff
@@ -36,15 +38,31 @@ object G8 {
     * possible to have other ValueF definitions which perform arbitrary logic given previously defined properties.
     */
   type ValueF = ResolvedProperties => String
-  
+
   def applyTemplate(default: String, resolved: ResolvedProperties): String = {
-      val group = STGroup('$', '$')
-      group.registerRenderer(renderer)
-      STHelper(group, default)
+    val group = STGroup('$', '$')
+    group.nativeGroup.setListener(new STErrorHandler)
+    group.registerRenderer(renderer)
+    STHelper(group, default)
       .setAttributes(resolved)
       .render()
   }
 
+  class STErrorHandler extends STErrorListener {
+    def compileTimeError(msg: STMessage) = {
+      throw new Exception("Template Compile Time Error")
```

Comment:
What are the common template errors? Are there useful error messages inside STMessage?